tensorzero-internal/src/endpoints/datasets.rs

```diff
@@ -249,9 +251,63 @@ pub async fn create_datapoint_handler(
     Ok(Json(CreateDatapointResponse {}))
 }
 
+#[instrument(name = "delete_datapoint", skip(app_state))]
+pub async fn delete_datapoint_handler(
+    State(app_state): AppState,
+    Path(path_params): Path<DeletePathParams>,
+) -> Result<Json<DeleteDatapointResponse>, Error> {
+    let datapoint = app_state.clickhouse_connection_info.run_query(
+        "SELECT * FROM {table_name:Identifier} WHERE dataset_name={dataset_name:String} and function_name={function_name:String} and id = {id:String} FORMAT JSONEachRow;".to_string(),
+        Some(&HashMap::from([
+            ("table_name", path_params.kind.table_name()),
+            ("function_name", path_params.function.as_str()),
+            ("dataset_name", path_params.dataset.as_str()),
+            ("id", path_params.id.to_string().as_str())
+        ]))).await?;
+
+    if datapoint.is_empty() {
+        return Err(Error::new(ErrorDetails::InvalidRequest {
+            message: format!("Datapoint not found with params {path_params:?}",),
+        }));
+    }
+
+    let mut datapoint_json: serde_json::Value = serde_json::from_str(&datapoint).map_err(|e| {
+        Error::new(ErrorDetails::Serialization {
+            message: format!("Failed to deserialize datapoint: {}", e),
+        })
+    })?;
+
+    // We delete datapoints by writing a new row (which ClickHouse will merge)
+    // with the 'is_deleted' and 'updated_at' fields modified.
+    datapoint_json["is_deleted"] = serde_json::Value::Bool(true);
+    datapoint_json["updated_at"] = dbg!(format!(
```

Comment:
you will want to remove the dbg! here

Reply:
also if there's a clippy we need to include to catch these we should prob add that too

Reply:
Opened https://github.com/tensorzero/tensorzero/pull/1224